foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java

```diff
@@ -2667,6 +2683,8 @@ public int appendParameterInternal(Call call, Writer writer, Object parameter) {
             } else if (dbValue instanceof java.time.OffsetTime ot) {
                 java.sql.Timestamp ts = java.sql.Timestamp.valueOf(java.time.LocalDateTime.of(java.time.LocalDate.ofEpochDay(0), ot.toLocalTime()));
                 appendTimestamp(ts, writer);
+            } else if (dbValue instanceof java.time.Year){
+                appendNumber((Number)dbValue, writer);
```

Comment:
just noticed that this cast is always going to fail

Reply:
Fixed in #2081